### PR TITLE
Makes mobs visually float when weightless 

### PR DESCRIPTION
**Pull request by AJCM-git**
## About the PR 
Very simple visualizer that gives some visual feedback
Also made pointing use the same animation

**Media**

https://user-images.githubusercontent.com/60196617/211247495-787076f8-ce3f-4db5-bfd3-b0ab1e9cdfaf.mp4

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: Makes mobs visually float when there is no gravity

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Does this work with mobs with offset sprites?

File: Content.Shared/Gravity/SharedGravitySystem.cs [hunk 1]
```diff
@@ -22,18 +23,28 @@ public bool IsWeightless(EntityUid uid, PhysicsComponent? body = null, Transform
             Resolve(uid, ref body, false);
 
             if ((body?.BodyType & (BodyType.Static | BodyType.Kinematic)) != 0)
+            {
+                _appearance.SetData(uid, GravityVisuals.Enabled, false);
                 return false;
+            }
```
Comment (metalgearsloth) [inline Content.Shared/Gravity/SharedGravitySystem.cs:29]: "IsX" functions should not be modifying state.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by EmoGarbage404** [comment]
Can this be generalized to work with various objects regardless of gravity? I might want to use this for anomalies.

File: Content.Shared/Gravity/SharedGravitySystem.cs (hunk 1 above)
Comment (AJCM-git) [reply Content.Shared/Gravity/SharedGravitySystem.cs]: done

File: Content.Shared/Gravity/SharedFloatingVisualsComponent.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Gravity;
+
+[RegisterComponent, NetworkedComponent]
+[Access(typeof(SharedFloatingVisualizerSystem))]
+public sealed class SharedFloatingVisualsComponent : Component
```
Comment (metalgearsloth) [inline Content.Shared/Gravity/SharedFloatingVisualsComponent.cs:8]: Don't prefix shared comps with shared.

File: Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    private void OnComponentStartup(EntityUid uid, SharedFloatingVisualsComponent component, ComponentStartup args)
+    {
+        if (!HasGravity(uid, component))
+            FloatAnimation(uid, component.Offset, component.AnimationKey, component.AnimationTime);
+    }
+
+    private void OnGravityChanged(ref GravityChangedEvent args)
+    {
+        foreach (var component in EntityQuery<SharedFloatingVisualsComponent>())
+        {
```
Comment (metalgearsloth) [inline Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs:50]: This doesn't consider paused entities.

File: Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            var uid = component.Owner;
+            component.HasGravity = args.HasGravity;
+            Dirty(component);
```
Comment (metalgearsloth) [inline Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs:53]: You should check the entity is even relevant for the grid (i.e. griduid / mapuid) before dirtying it which is expensive.

File: Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+            if (!args.HasGravity)
+                FloatAnimation(uid, component.Offset, component.AnimationKey, component.AnimationTime);
+        }
+    }
+
+    private void OnEntParentChanged(EntityUid uid, SharedFloatingVisualsComponent component, ref EntParentChangedMessage args)
+    {
+        var gridUid = args.Transform.GridUid;
+        if (!HasGravity(uid, component, gridUid))
```
Comment (metalgearsloth) [inline Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs:63]: Maps without grids can have gravity too.

**Review by metalgearsloth — DISMISSED** [review]

File: Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs (hunk 5 above)
Comment (AJCM-git) [reply Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs]: yeah im passing the transform and using `IsWeigthless` now

**Pull request merged**